File: api_control_neumaticos/Models/Neumatico.cs
using System;
using System.Collections.Generic;

namespace api_control_neumaticos.Models;

public partial class Neumatico
{
    public int ID_NEUMATICO { get; set; }
    public int CODIGO { get; set; }
    public int UBICACION { get; set; }

    // Relación con Movil, Movil puede ser nulo
    public int? ID_MOVIL { get; set; }
    public virtual Movil? MOVIL_ASIGNADO { get; set; }

    // Relación con Bodega
    public int ID_BODEGA { get; set; }
    public virtual required Bodega BODEGA { get; set; }

    public DateTime FECHA_INGRESO { get; set; }
    public DateTime? FECHA_SALIDA { get; set; }
    public int ESTADO { get; set; }
    public int KM_TOTAL { get; set; }
    public int TIPO_NEUMATICO { get; set; }
}

File: flutter_control_neumaticos/lib/models/neumatico.dart
class Neumatico {
  final String? idNeumatico;
  final String codigo;
  final int? ubicacion;
  final int? idMovil;
  final int idBodega;
  final String fechaIngreso;
  final int estado;
  final int kmTotal;
  final int? tipoNeumatico;

  Neumatico({
    this.idNeumatico,
    required this.codigo,
    this.ubicacion,
    this.idMovil,
    required this.idBodega,
    required this.fechaIngreso,
    required this.estado,
    required this.kmTotal,
    this.tipoNeumatico,
  });

  Map<String, dynamic> toJson() => {
        "ID_NEUMATICO": idNeumatico,
        "CODIGO": codigo,
        "UBICACION": ubicacion,
        "ID_MOVIL": idMovil,
        "ID_BODEGA": idBodega,
        "FECHA_INGRESO": fechaIngreso,
        "ESTADO": estado,
        "KM_TOTAL": kmTotal,
        "TIPO_NEUMATICO": tipoNeumatico,
      };
  factory Neumatico.fromJson(Map<String, dynamic> json)
  {
    return Neumatico(
      idNeumatico: json['ID_NEUMATICO'],
      codigo: json['CODIGO'],
      ubicacion: json['UBICACION'],
      idMovil: json['ID_MOVIL'],
      idBodega: json['ID_BODEGA'],
      fechaIngreso: json['FECHA_INGRESO'],
      estado: json['ESTADO'],
      kmTotal: json['KM_TOTAL'],
      tipoNeumatico: json['TIPO_NEUMATICO'],
    );
  }
}
